include/boost/unordered/detail/foa.hpp

```diff
@@ -1163,11 +1171,20 @@ constexpr static float const mlf = 0.875f;
  *     a copyable const std::string&&. foa::table::insert is extended to accept
  *     both init_type and value_type references.
  *
- *   - TypePolicy::move(element_type&) returns a temporary object for value
- *     transfer on rehashing, move copy/assignment, and merge. For flat map, this
- *     object is a std::pair<Key&&,T&&>, which is generally cheaper to move
- *     than std::pair<const Key,T>&& because of the constness in Key. For
- *     node-based tables, this is used to transfer ownership of pointer.
+ *   - TypePolicy::construct and TypePolicy::destroy are used for the
+ *     construction and destruction of the internal types: value_type, init_type
+ *     and element_type. For flat containers, these are often all synonyms for
+ *     each other but for the node-based map, each one is a distinct type. These
+ *     are used for allocator-aware construction and destruction of the types
```

Comment:
It is not the case that for flat containers `value_type`, `init_type` and `element_type` are the same. In fact:
* For set containers, `init_type` and `value_type` are the same, for map containers they're not.
* For flat containers, `value_type` and `element_type` are the same, for node containers they're not,

Anyway, I'd just remove the "For flat containers [...] during insertion" part altogether, as it's redundant with the rest of the information already provided elsewhere in the comment.